src/test/java/io/jenkins/plugins/grading/AutoGraderITest.java

```diff
@@ -55,20 +74,261 @@ public void shouldAbortBuildSinceNoTestActionHasBeenRegistered() {
     public void shouldCountCheckStyleWarnings() {
         WorkflowJob job = createPipelineWithWorkspaceFiles("checkstyle.xml");
 
-        configureScanner(job, "checkstyle", "{\"analysis\":{\"maxScore\":100,\"errorImpact\":-10,\"highImpact\":-5,\"normalImpact\":-2,\"lowImpact\":-1}}");
+        configureScanner(job, "checkstyle", ANALYSIS_CONFIGURATION);
         Run<?, ?> baseline = buildSuccessfully(job);
 
         assertThat(getConsoleLog(baseline)).contains("[Autograding] Grading static analysis results for CheckStyle");
         assertThat(getConsoleLog(baseline)).contains("[Autograding] -> Score -60 (warnings distribution err:6, high:0, normal:0, low:0)");
         assertThat(getConsoleLog(baseline)).contains("[Autograding] Total score for static analysis results: 40");
 
-        List<AutoGradingBuildAction> actions = baseline.getActions(AutoGradingBuildAction.class);
+        checkstyleAssertions(baseline);
+    }
+
+    @Test
+    public void shouldCountCheckstyleFreeStyle(){
+        FreeStyleProject project = createFreeStyleProjectWithWorkspaceFiles("checkstyle.xml");
+        IssuesRecorder recorder = new IssuesRecorder();
+        CheckStyle checkStyle = new CheckStyle();
+        checkStyle.setPattern("checkstyle.xml");
+        recorder.setTools(checkStyle);
+        project.getPublishersList().add(recorder);
+        project.getPublishersList().add(new AutoGrader(ANALYSIS_CONFIGURATION));
+        Run<?, ?> run = buildSuccessfully(project);
+
+        checkstyleAssertions(run);
```

Comment:
Methoden sollten ein Verb als Namen haben:
```suggestion
        assertCheckStyleResults(run);
```